dd-trace-ot/src/main/java/datadog/opentracing/DDSpan.java

```diff
@@ -137,83 +133,43 @@ public void setErrorMeta(final Throwable error) {
   }
 
   private boolean extractError(final Map<String, ?> map) {
-    if (map.get(ERROR_OBJECT) instanceof Throwable) {
-      final Throwable error = (Throwable) map.get(ERROR_OBJECT);
+    if (map.get(LogFields.ERROR_OBJECT) instanceof Throwable) {
+      final Throwable error = (Throwable) map.get(LogFields.ERROR_OBJECT);
       setErrorMeta(error);
       return true;
     }
     return false;
   }
 
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#setTag(java.lang.String, java.lang.String)
-   */
   @Override
   public final DDSpan setTag(final String tag, final String value) {
-    context().setTag(tag, (Object) value);
+    context().setTag(tag, value);
     return this;
   }
 
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#setTag(java.lang.String, boolean)
-   */
   @Override
   public final DDSpan setTag(final String tag, final boolean value) {
-    context().setTag(tag, (Object) value);
+    context().setTag(tag, value);
     return this;
   }
 
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#setTag(java.lang.String, java.lang.Number)
-   */
   @Override
   public final DDSpan setTag(final String tag, final Number value) {
-    context().setTag(tag, (Object) value);
+    context().setTag(tag, value);
     return this;
   }
 
-  @Override
-  public <T> Span setTag(final Tag<T> tag, final T value) {
-    context().setTag(tag.getKey(), value);
-    return this;
-  }
-
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#context()
-   */
   @Override
   public final DDSpanContext context() {
     return context;
   }
 
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#getBaggageItem(java.lang.String)
-   */
-  @Override
-  public final String getBaggageItem(final String key) {
-    return null;
-  }
-
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#setBaggageItem(java.lang.String, java.lang.String)
-   */
-  @Override
-  public final DDSpan setBaggageItem(final String key, final String value) {
-    return this;
-  }
-
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#setOperationName(java.lang.String)
-   */
   @Override
   public final DDSpan setOperationName(final String operationName) {
     context().setOperationName(operationName);
     return this;
   }
 
-  /* (non-Javadoc)
-   * @see io.opentracing.BaseSpan#log(java.util.Map)
-   */
   @Override
   public final DDSpan log(final Map<String, ?> map) {
     if (!extractError(map)) {
```

Comment:
I'd suggest moving this logic up to the caller in `AgentTracerImpl` and removing the `log` method.